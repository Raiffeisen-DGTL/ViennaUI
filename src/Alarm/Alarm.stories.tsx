import React from 'react';
import { Story, Meta } from 'storybook';
import { Alarm, AlarmProps } from './Alarm';

export default {
    title: 'Development/Alarm',
    component: Alarm,
} as Meta;

export const Overview: Story<AlarmProps> = (args) => {
    return <Alarm design='neutral' {...args} />;
};

export const Designs: Story<AlarmProps> = (args) => {
    return (
        <>
            <Alarm design='critical' />
            <Alarm design='warning' />
            <Alarm design='success' />
            <Alarm design='accent' />
            <Alarm design='neutral' />
            <Alarm design='disabled' />
        </>
    );
};

Designs.storyName = 'Дизайн';
