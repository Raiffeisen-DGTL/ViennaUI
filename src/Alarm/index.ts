export { type AlarmProps, type AlarmDesign, Alarm } from './Alarm';
