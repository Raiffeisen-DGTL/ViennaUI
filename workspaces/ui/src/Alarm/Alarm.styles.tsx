import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const presets = getPresets('alarm', {
    base: null,
    design: 'design',
});

export const Box = styled.div<any>`
    position: ${(props) => props.position};
    display: inline-flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;

    ${presets.base}
    ${presets.design}

    ${(props) =>
        props.position === 'absolute' &&
        css`
            top: ${(props: any) => props.top || 'auto'};
            bottom: ${(props: any) => props.bottom || 'auto'};
            left: ${(props: any) => props.left || 'auto'};
            right: ${(props: any) => props.right || 'auto'};
        `}
`;
