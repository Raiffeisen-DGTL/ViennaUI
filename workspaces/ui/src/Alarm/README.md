# Alarm

Короткие уведомления. Используются для основной навигации, раздела пуш-уведомлений (звоночек в шапке).


## Импорт

```
import { Alarm } from 'vienna-ui';
```

## Свойства / Props

Свойства наследуются от [HTMLAttributes<HTMLDivElement>](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L1746)

Prop | Type | Default | Description
--- | --- | --- | ---
design | "critical" \| "warning" \| "success" \| "accent" \| "neutral" \| "disabled" \| undefined | false |
position | "relative" \| "absolute" \| undefined | false |
top | string \| undefined | false | Свойство для вертикального позиционирования. Зависит от свойства position
bottom | string \| undefined | false | Свойство для вертикального позиционирования. Зависит от свойства position
left | string \| undefined | false | Свойство для горизонтального позиционирования. Зависит от свойства position
right | string \| undefined | false | Свойство для горизонтального позиционирования. Зависит от свойства position

## Использование

```
<Alarm design='success' />
```
