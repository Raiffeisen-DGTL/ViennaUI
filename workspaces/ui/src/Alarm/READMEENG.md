# Alarm

Short notifications. It's used for main navigation and push-notification section (bell on the header).


## Imports

```
import { Alarm } from 'vienna-ui';
```

## Props

Prop | Type | Default | Description
--- | --- | --- | ---
design | "critical" \| "warning" \| "success" \| "accent" \| "neutral" \| "disabled" \| undefined | false |
position | "relative" \| "absolute" \| undefined | false |
top | string \| undefined | false | The top property participates in specifying the vertical position of a positioned element. Depends on position property
bottom | string \| undefined | false | The bottom property participates in specifying the vertical position of a positioned element. Depends on position property
left | string \| undefined | false | The left property participates in specifying the horizontal position of a positioned element. Depends on position property
right | string \| undefined | false | The right property participates in specifying the horizontal position of a positioned element. Depends on position property

## Usage

```
<Alarm design='success' />
```
