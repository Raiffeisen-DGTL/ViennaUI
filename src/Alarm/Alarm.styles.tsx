import styled, { css } from 'styled-components';
import { getPresets } from '../Utils/styling';
import { AlarmDesign } from './Alarm';

const { base, design } = getPresets('alarm', {
    base: null,
    design: '$design',
});

export interface PropsBox {
    $design: AlarmDesign;
    $position?: 'relative' | 'absolute';
    $top?: string;
    $bottom?: string;
    $left?: string;
    $right?: string;
}
export const Box = styled.div<PropsBox>`
    position: ${({ $position }) => $position};
    display: inline-flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;

    ${base}
    ${design}

    ${({ $position, $top, $bottom, $left, $right }) =>
        $position === 'absolute' &&
        css`
            top: ${$top ?? 'auto'};
            bottom: ${$bottom ?? 'auto'};
            left: ${$left ?? 'auto'};
            right: ${$right ?? 'auto'};
        `}
`;
