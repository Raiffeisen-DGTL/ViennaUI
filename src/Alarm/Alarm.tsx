import React, { forwardRef } from 'react';
import { Box, PropsBox } from './Alarm.styles';
import { BoxStyled } from '../Utils/styled';

export type AlarmDesign = 'critical' | 'warning' | 'success' | 'accent' | 'neutral' | 'disabled';

export interface AlarmProps extends BoxStyled<typeof Box, PropsBox> {
    design: PropsBox['$design'];
    position?: PropsBox['$position'];
    top?: PropsBox['$top'];
    bottom?: PropsBox['$bottom'];
    left?: PropsBox['$left'];
    right?: PropsBox['$right'];
}

export const Alarm = forwardRef<HTMLDivElement, AlarmProps>(
    ({ children, design, position, top, bottom, left, right, ...attrs }, ref) => {
        return (
            <Box
                {...(attrs as {})}
                ref={ref}
                $design={design}
                $position={position}
                $top={top}
                $bottom={bottom}
                $left={left}
                $right={right}>
                {children}
            </Box>
        );
    }
);

Alarm.displayName = 'Alarm';
