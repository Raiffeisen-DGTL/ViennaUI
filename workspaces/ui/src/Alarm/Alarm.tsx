import React, { HTMLAttributes } from 'react';
import { Box } from './Alarm.styles';

export interface AlarmProps extends HTMLAttributes<HTMLDivElement> {
    design?: 'critical' | 'warning' | 'success' | 'accent' | 'neutral' | 'disabled';
    position?: 'relative' | 'absolute';
    top?: string;
    bottom?: string;
    left?: string;
    right?: string;
}

export const Alarm: React.FC<AlarmProps> = (props): JSX.Element => {
    return <Box {...props} />;
};

Alarm.displayName = 'Alarm';
Alarm.defaultProps = {
    design: 'neutral',
    position: 'relative',
};

export default Alarm;
