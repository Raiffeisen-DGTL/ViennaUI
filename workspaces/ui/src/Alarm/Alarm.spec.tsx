import React from 'react';
import Alarm, { AlarmProps } from './Alarm';

test('Alarm', async () => {
    const snap = snapshot.render(<Alarm />);
    expect(snap).toMatchSnapshot();
});

test('Alarm w/ design', async () => {
    const components: JSX.Element[] = [];

    const designs: AlarmProps['design'][] = ['critical', 'warning', 'success', 'accent', 'neutral', 'disabled'];
    designs.forEach((design) => components.push(<Alarm design={design} />));

    const snap = snapshot.render(components);
    expect(snap).toMatchSnapshot();
});

test('Alarm w/ position', () => {
    const positions: AlarmProps['position'][] = ['relative', 'absolute'];

    positions.forEach((position) => {
        const snap = snapshot.render(<Alarm position={position} top='10px' bottom='10px' left='10px' right='10px' />);
        expect(snap).toMatchSnapshot(position);
    });
});

test('Alarm w/ any props', () => {
    const props = { className: 'className', id: 'alarm', title: 'title', onClick: () => 'Alarm' };
    const component = <Alarm {...props} />;

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});
