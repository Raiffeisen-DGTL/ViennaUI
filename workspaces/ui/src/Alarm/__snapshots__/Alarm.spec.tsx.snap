// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alarm 1`] = `
<div
  class="Alarmstyles__Box-sc-1k22vn9-0 jlXDJL"
/>
`;

exports[`Alarm w/ any props 1`] = `
<div
  class="Alarmstyles__Box-sc-1k22vn9-0 jlXDJL className"
  id="alarm"
  title="title"
/>
`;

exports[`Alarm w/ design 1`] = `
Array [
  <div
    class="Alarmstyles__Box-sc-1k22vn9-0 HfwfI"
  />,
  <div
    class="Alarmstyles__Box-sc-1k22vn9-0 dpCONz"
  />,
  <div
    class="Alarmstyles__Box-sc-1k22vn9-0 eBikTp"
  />,
  <div
    class="Alarmstyles__Box-sc-1k22vn9-0 hvWWms"
  />,
  <div
    class="Alarmstyles__Box-sc-1k22vn9-0 jlXDJL"
  />,
  <div
    class="Alarmstyles__Box-sc-1k22vn9-0 cyKRRz"
  />,
]
`;

exports[`Alarm w/ position: absolute 1`] = `
<div
  class="Alarmstyles__Box-sc-1k22vn9-0 FgHtH"
/>
`;

exports[`Alarm w/ position: relative 1`] = `
<div
  class="Alarmstyles__Box-sc-1k22vn9-0 jlXDJL"
/>
`;
